client/js/dashboard.js

```diff
@@ -108,6 +108,8 @@ if (Element && !Element.prototype.matches) {
             console.error(err);
             ErrorIndicator.show();
         });
+    }).catch(function(e) {
+        ErrorIndicator.show();
```

Comment:
I think we can make this cleaner:

1. Return the promise from `widgetsReady.then()` (line 67).
2. Remove the catch for that line, lines 107-110.
3. Add `console.error(err)` to your new code.

This way, there's only one error handler rather than 2.